### DISCUSSION TITLE
Appending Files into an Existing Zip file in S3

### DISCUSSION THREAD

**Discussion by jey-ehstools** [RaRe-Technologies/smart_open#845] [Q&A] [answered]
I want to append additional files into an existing Zip file in S3, currently im getting error if try use 'a' in my zipfile.ZipFile(fout, 'a'). Is it possible, if yes, what am I doing wrong?

chunk_size_mb = 256
with smart_open.open(output_path, 'rb', transport_params={'client': session}) as fout:
    with zipfile.ZipFile(fout, 'a') as zip:
        for file_key in prefix:
            for key, content in s3.iter_bucket(source_bucket_name, prefix = file_key, workers=2, **session_kwargs):
                s3_object = session.get_object(Bucket=source_bucket_name, Key=key)
                with zip.open(key, 'w') as zip_archive_file:
                    for chunk in s3_object['Body'].iter_chunks(chunk_size=1024 * chunk_size_mb):
                        zip_archive_file.write(chunk)

**Comment by ddelange** [comment, marked as answer]
AWS S3 blobs are immutable: if you want to append, you have to upload (overwrite) a new blob
